.github/workflows/pr-validate.yaml

```diff
@@ -34,22 +34,38 @@ jobs:
         with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
           script: |
-            const url = `${process.env.server_url}/${process.env.repo}/actions/runs/${process.env.run_id}`
-            const { data: pull } = await github.rest.pulls.get({
-              ...context.repo,
-              pull_number: process.env.number
-            });
-            console.log('retrieved pull request')
-            
-            const ref = pull.head.sha;
-            const { data: result } = await github.rest.checks.create({
-              ...context.repo,
-              name: process.env.validate_job_name,
-              head_sha: ref,
-              status: 'completed',
-              conclusion: process.env.conclusion,
-              details_url: url,
-            });
-            console.log('created check')
-            
-            return result;
+            if (context.eventName == 'pull_request') {
+              const url = `${process.env.server_url}/${process.env.repo}/actions/runs/${process.env.run_id}`
+              const { data: pull } = await github.rest.pulls.get({
+                ...context.repo,
+                pull_number: process.env.number
+              });
+              console.log('retrieved pull request')
+              
+              const ref = pull.head.sha;
+              const { data: result } = await github.rest.checks.create({
+                ...context.repo,
+                name: process.env.validate_job_name,
+                head_sha: ref,
+                status: 'completed',
+                conclusion: process.env.conclusion,
+                details_url: url,
+              });
+              console.log('created check')
+              return result;
+            }
+            if (context.eventName == 'schedule') {
+              const url = `${process.env.server_url}/${process.env.repo}/actions/runs/${process.env.run_id}`
```

Comment:
It seems that the only difference is that the head_sha is either the pull request sha, or the context.sha. In order to avoid a lot of duplicated code, I think you can pull the `github.rest.checks.create` call to outside the if statements.